times_reader/excel.py

```diff
@@ -131,19 +131,24 @@ def extract_table(
             table_df = DataFrame(df.iloc[header_row, start_col:end_col])
             table_df.columns = ["VALUE"]
         else:
-            table_df = DataFrame(df.iloc[header_row + 1 : end_row, start_col:end_col])
+            table_df = df.iloc[header_row + 1 : end_row, start_col:end_col]
             # Make all columns names strings as some are integers e.g. years
             table_df.columns = [str(x) for x in df.iloc[header_row, start_col:end_col]]
 
     table_df.reset_index(drop=True, inplace=True)
-    table_df = table_df.applymap(
-        lambda cell: cell if not isinstance(cell, float) else utils.round_sig(cell, 15)
-    )
+
+    # Don't use applymap because it can convert ints to floats
+    # https://pandas.pydata.org/pandas-docs/stable/user_guide/gotchas.html#gotchas-intna
```

Comment:
Aha, good to know!

Reply:
I think there are probably other places where integers, mainly years, are being converted to floats and NaNs